src/app/static/src/app/store/downloadResults/actions.ts

```diff
@@ -0,0 +1,103 @@
+
+export interface DownloadResultsActions {
+    downloadSummary: (store : ActionContext<DownloadResultsState, RootState>) => void
+    downloadSpectrum: (store : ActionContext<DownloadResultsState, RootState>) => void
+    downloadCoarseOutput: (store : ActionContext<DownloadResultsState, RootState>) => void
+    poll: (store: ActionContext<DownloadResultsState, RootState>, downloadType: string) => void
+}
+
+export const actions: ActionTree<DownloadResultsState, RootState> & DownloadResultsActions = {
+
+    async downloadCoarseOutput(context) {
+        const {dispatch, rootState} = context
+        const calibrateId = rootState.modelCalibrate.calibrateId
+
+        const response = await api<DownloadResultsMutation, DownloadResultsMutation>(context)
+            .withSuccess(DownloadResultsMutation.CoarseOutputDownloadStarted)
+            .withError(DownloadResultsMutation.CoarseOutputError)
+            .get(`download/submit/coarse-output/${calibrateId}`)
+
+        if (response) {
+            await dispatch("poll", DOWNLOAD_TYPE.COARSE)
+        }
+    },
+
+    async downloadSummary(context) {
+        const {dispatch, rootState} = context
+        const calibrateId = rootState.modelCalibrate.calibrateId
+
+        const response = await api<DownloadResultsMutation, DownloadResultsMutation>(context)
+            .withSuccess(DownloadResultsMutation.SummaryDownloadStarted)
+            .withError(DownloadResultsMutation.SummaryError)
+            .get(`download/submit/summary/${calibrateId}`)
+
+        if (response) {
+            await dispatch("poll", DOWNLOAD_TYPE.SUMMARY)
+        }
+    },
+
+    async downloadSpectrum(context) {
+        const {dispatch, rootState} = context
+        const calibrateId = rootState.modelCalibrate.calibrateId
+
+        const response = await api<DownloadResultsMutation, DownloadResultsMutation>(context)
+            .withSuccess(DownloadResultsMutation.SpectrumDownloadStarted)
+            .withError(DownloadResultsMutation.SpectrumError)
+            .get(`download/submit/spectrum/${calibrateId}`)
+
+        if (response) {
+            await dispatch("poll", DOWNLOAD_TYPE.SPECTRUM)
+        }
+    },
+
+    async poll(context, downloadType) {
+        const {commit} = context;
+        const id = setInterval(() => {
+
+            if(downloadType === DOWNLOAD_TYPE.SPECTRUM) {
+                getSpectrumDownloadStatus(context)
+            }
+            else if(downloadType === DOWNLOAD_TYPE.COARSE) {
+                getCoarseOutputDownloadStatus(context)
+            }
+            else if (downloadType === DOWNLOAD_TYPE.SUMMARY) {
+                getSummaryDownloadStatus(context)
+            }
+        }, 2000);
+
+        commit({type: "PollingStatusStarted", payload: {pollId: id, downloadType: downloadType}});
+    },
+};
+
+export const getSummaryDownloadStatus = async function (context: ActionContext<DownloadResultsState, RootState>) {
```

Comment:
Would be good to add a return type to this function. And same for the other two.

Reply:
I am not sure why we need a return type right now, what are you thinking ?

Reply:
Was thinking that with TypeScript it's good for functions to have return types. But admittedly not something we do everywhere. And just realised that in this case we don't care about the result, only the mutations.